### PR TITLE
Add yaml lint and shellcheck tests

### PR DESCRIPTION
**Pull request by loafyloaf**
This commit starts the process of bringing this repo in line with
the standard CI system by introducing yaml lint and shellcheck
tests.

### CODE REVIEW

**Comment by loafyloaf** [comment]
oops

**Comment by Tomcli** [comment]
Looks like it failed with pip install. Any idea?

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,16 @@
+---
 language: bash
 script: true
 sudo: required
-before_install:
-- bash scripts/install.sh
 group: stable
 dist: precise
 os: linux
```
Comment (mlangbehn) [inline .travis.yml]: I would suggest changing this first bit to:

```yaml
---
language: python
python: 2.7
cache: pip
```

Comment (mlangbehn) [reply .travis.yml]: We can add the rest to this later.

**Review by mlangbehn — CHANGES_REQUESTED** [review]

File: test-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+flake8
```
Comment (mlangbehn) [inline test-requirements.txt]: We aren't using flake8 in CI for this repo, so let's remove it from here.

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,22 @@
-language: bash
-script: true
+---
+language: python
+python: 2.7
+cache: pip
 sudo: required
```
Comment (mlangbehn) [inline .travis.yml:5]: Unless we start using docker in Travis CI, we don't actually need sudo. If we remove this line, builds will speed up.

File: .travis.yml [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
+
 before_install:
-- bash scripts/install.sh
-group: stable
-dist: precise
-os: linux
+  - pip install -U -r test-requirements.txt
+
+install:
+  - "./scripts/install.sh"
```
Comment (mlangbehn) [inline .travis.yml:18]: This line is still failing in Travis CI. Please run `./tests/test-shellcheck.sh` locally, and address any issues that arise.

Feel free to reach out to me with any questions.

**Review by mlangbehn — DISMISSED** [review]

File: .travis.yml (hunk 3 above)
Comment (loafyloaf) [reply .travis.yml:5]: the way install.sh is currently written, sudo is required.  it fails horribally without it

File: .travis.yml (hunk 4 above)
Comment (loafyloaf) [reply .travis.yml:18]: working now

File: .travis.yml (hunk 1 above)
Comment (loafyloaf) [reply .travis.yml]: changes made

**Comment by mlangbehn** [comment]
Current CI failures are unrelated to this change. Merging now; @loafyloaf will open another pull request to address the rest of the CI system.

**Review by mlangbehn — APPROVED** [review]

**Pull request merged**